src/utils/filterProducts.js

```diff
@@ -0,0 +1,32 @@
+export const filterProducts = (products, filterState) => {
+  const { sortBy, category, price, rating } = filterState;
+  let filteredProducts = [...products];
+
+  // filter by price b/w lowest and selected value
+  filteredProducts = products.filter((product) => product.price <= price);
+
+  //   filter by rating higher than selected value
+  filteredProducts = filteredProducts.filter(
+    (product) => product.rating >= rating
+  );
+
+  // sort by filters
+  if (sortBy === "LOW_TO_HIGH") {
+    filteredProducts = [...filteredProducts].sort((a, b) => a.price - b.price);
+  }
+  if (sortBy === "HIGH_TO_LOW") {
+    filteredProducts = [...filteredProducts].sort((a, b) => b.price - a.price);
+  }
+
```

Comment:
both of the below examples do the same thing
```js
filteredProducts = [...filteredProducts].sort((a, b) => a.price - b.price);
```
```js
filteredProducts.sort((a, b) => a.price - b.price);
```
filteredPrducts is not your state and you have already created a new array of products when you applied `.filter()` a few lines above, so you can directly sort without destructuring it first.

Reply:
Since sort mutates state, I had spread it. But, your point makes sense. Will make this change.